templates/tfengine/recipes/deployment.hcl

```diff
@@ -155,7 +155,11 @@ template "terraform" {
   {{end}}
 }
 
-{{if has . "terraform_addons.vars"}}
+// The key "terraform_addons" could be flattened at one level above
+// where "deployment.hcl" recipe is used, e.g. "project.hcl".
+// "deployment.hcl" can also be directly used, in which case "terraform_addons"
+// is not flattened.
+{{if or (has . "vars") (has . "terraform_addons.vars")}}
```

Comment:
Will this work? If the first case is true (. has "vars"), then there's no "terraform_addons" key to flatten, so does it just get passed down directly?

Reply:
Yes. Note this is in `deployment.hcl` and values here will be passed down to each components (e.g. variables, states, etc).

There are two cases where this part will be reached:

1. `deployment.hcl` is called from `project.hcl`, and `terraform_addons` key is flattened/removed there. In that case, the first case in this `or` statement will be hit, the proper component will still be added, and the additional `flatten` statement below will not be hit as the if condition there will be false.

2. `deployment.hcl` is called directly, e.g. in `modules/team.hcl` in the kubernetes template. In that case, second case in this `or` statement will be hit, and the `flatten` operation below will take effect.